test/unit/L2/l2Migrator.test.ts

```diff
@@ -81,47 +111,205 @@ describe('L2Migrator', function() {
     it('reverts if msg.sender is not L1Migrator L2 alias', async () => {
       // msg.sender = some invalid address
       let tx = l2Migrator.connect(notL1MigratorEOA).finalizeMigrateDelegator(
-          mockMigrateDelegatorParams,
+          mockMigrateDelegatorParams(),
       );
       await expect(tx).to.revertedWith('ONLY_COUNTERPART_GATEWAY');
 
       // msg.sender = L1Migrator (no alias)
       tx = l2Migrator.connect(mockL1MigratorEOA).finalizeMigrateDelegator(
-          mockMigrateDelegatorParams,
+          mockMigrateDelegatorParams(),
       );
       await expect(tx).to.revertedWith('ONLY_COUNTERPART_GATEWAY');
     });
 
-    it('finalizes migration', async () => {
+    it('reverts when l1Addr already migrated', async () => {
+      await l2Migrator.connect(mockL1MigratorL2AliasEOA).finalizeMigrateDelegator(
+          mockMigrateDelegatorParams(),
+      );
+
       const tx = l2Migrator.connect(mockL1MigratorL2AliasEOA).finalizeMigrateDelegator(
-          mockMigrateDelegatorParams,
+          mockMigrateDelegatorParams(),
       );
+      await expect(tx).to.revertedWith('L2Migrator#finalizeMigrateDelegator: ALREADY_MIGRATED');
+    });
 
-      await expect(tx)
-          .to.emit(l2Migrator, 'MigrateDelegatorFinalized');
-      // The assertion below does not work until https://github.com/EthWorks/Waffle/issues/245 is fixed
-      // .withArgs(seqNo, mockMigrateDelegatorParams)
+    it('reverts if fee transfer fails', async () => {
+      const params = mockMigrateDelegatorParams();
+      params.fees = 300;
```

Comment:
#minor
Since there's no nesting, can just use `params = {...mockMigrateDelegatorParams}` to create a clone and modify keys.